Improve UX clarity between "Sync Now" and "Import all Products from Square" buttons

m-muhsin opened the pull request:
### All Submissions:

* [x] Does your code follow the [WooCommerce Sniffs](https://github.com/woocommerce/woocommerce-sniffs/) variant of WordPress coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Will this change require new documentation or changes to existing documentation?

---

### Changes proposed in this Pull Request:

This PR improves clarity between the "Sync Now" and "Import all Products from Square" actions to reduce merchant confusion.

**Problem:** Merchants were confused because both buttons lead to similar-looking pages, but they do different things:
- "Sync Now" only updates *existing* linked products
- "Import all Products from Square" brings *new* products from Square into WooCommerce

This caused support cases where merchants couldn't understand why new products added in Square weren't appearing after using "Sync Now".

**Solution:** Added descriptive helper text in key locations to clarify the difference:

1. **Update tab:** Added helper text below the "Sync Now" button and a warning notice in the sync modal explaining it only updates existing synced products (when Square is the system of record)

2. **Settings tab (PHP):** Added description text below the "Import all Products from Square" button explaining how it differs from "Sync Now"

3. **Settings tab (React):** Added the same description text and localized the import modal title

Closes #447  
Closes https://linear.app/a8c/issue/SQUARE-225/improve-ux-clarity-between-sync-now-and-import-all-products-from.

### Steps to test the changes in this Pull Request:

1. Set Square as the System of Record in WooCommerce > Settings > Square
2. Navigate to WooCommerce > Settings > Square > Update tab
3. Verify the "Sync Now" button now shows helper text: "Updates existing synced products only..."
4. Click "Sync Now" and verify the modal includes a yellow warning note about scope
5. Navigate to WooCommerce > Settings > Square > Settings tab
6. Verify the "Import all Products from Square" button has description text below it
7. Test both the React-based settings (new experience) and legacy PHP settings

### Changelog entry

> Add - Helper text and notices to clarify the difference between "Sync Now" and "Import all Products from Square".

Commit:
Enhance product import and sync messaging in WooCommerce Square integration.

Updated descriptions to clarify the distinction between importing new products and syncing existing ones. Added informational notices in the admin settings and sync pages to guide users effectively.

iamdharmesh commented on includes/Admin/Settings_Page.php:
```diff
@@ -169,12 +169,16 @@ private static function output_import_products_modal_template() {
+							<p style="background: #f0f6fc; border-left: 4px solid #2271b1; padding: 10px 12px; margin-bottom: 16px;">
+								<strong><?php esc_html_e( 'This is for importing new products.', 'woocommerce-square' ); ?></strong>
+								<?php esc_html_e( 'To update existing synced products, use "Sync Now" on the Update tab instead.', 'woocommerce-square' ); ?>
+							</p>
```
Import products option also provide a way to update existing products while doing import, So, this is not accurate.

<img width="1090" height="627" alt="Image" src="https://github.com/user-attachments/assets/1153d54f-3b52-4e41-ac8f-75b4b9d2791c" />

iamdharmesh commented on includes/Admin/Settings_Page.php:
```diff
@@ -169,12 +169,16 @@ private static function output_import_products_modal_template() {
 					<section class="wc-backbone-modal-main" role="main">
 						<header class="wc-backbone-modal-header">
-							<h1><?php esc_html_e( 'Import Products From Square', 'woocommerce-square' ); ?></h1>
+							<h1><?php esc_html_e( 'Import New Products From Square', 'woocommerce-square' ); ?></h1>
```
Better to keep this as it is, As mentioned below, import products option also provide a way to update existing products.

iamdharmesh commented on includes/Admin/Sync_Page.php:
```diff
@@ -250,8 +251,11 @@ private static function output_sync_status() {
 						<div id="wc-square-sync-progress-spinner" class="spinner" style="float:none; <?php echo $sync_in_progress ? 'visibility:visible;' : ''; ?>"></div>
+						<p class="description" style="margin-top: 8px; font-style: italic; color: #646970;">
+							<?php esc_html_e( 'Updates existing synced products only. Does not import new products from Square.', 'woocommerce-square' ); ?>
+						</p>
```
This should display only when Square is SOR, Apply to every change.

iamdharmesh commented on includes/Admin/Sync_Page.php:
```diff
@@ -49,6 +49,7 @@ public static function output() {
 			<h2><?php esc_html_e( 'Update', 'woocommerce-square' ); ?></h2>
 			<?php self::output_system_record_of_data_info(); ?>
 			<?php self::output_sync_status(); ?>
+			<?php self::output_import_products_notice(); ?>
```
As we are already showing help text below sync now button and notice in popup as well. I don't think we need to display this notice. Better to keep it minimal.

Commit:
Revert to: Import Products From Square

Commit:
Refactor product import and sync messaging in admin pages.

Removed redundant notice about importing new products from Square in the Settings Page. Updated the Sync Page to conditionally display the description for syncing existing products, enhancing clarity for users.

iamdharmesh commented on includes/Admin/Sync_Page.php:
```diff
@@ -260,6 +265,39 @@ class="button button-large"
+	 *
+	 * @since 4.8.0
+	 */
+	private static function output_import_products_notice() {
```
Maybe we should remove this as we are no longer using this.

iamdharmesh commented on includes/Admin/Sync_Page.php:
```diff
@@ -348,13 +386,17 @@ private static function output_sync_with_square_modal_template() {
 					<section class="wc-backbone-modal-main" role="main">
 						<header class="wc-backbone-modal-header">
-							<h1><?php esc_html_e( 'Sync products with Square', 'woocommerce-square' ); ?></h1>
+							<h1><?php esc_html_e( 'Sync existing products with Square', 'woocommerce-square' ); ?></h1>
```
```suggestion
							<h1><?php esc_html_e( 'Sync products with Square', 'woocommerce-square' ); ?></h1>
```

Please see comment below.

iamdharmesh commented on includes/Admin/Sync_Page.php:
```diff
@@ -348,13 +386,17 @@ private static function output_sync_with_square_modal_template() {
 						</header>
 						<article>
 							<?php $square_settings = wc_square()->get_settings_handler(); ?>
+							<p style="background: #fcf9e8; border-left: 4px solid #dba617; padding: 10px 12px; margin-bottom: 16px;">
+								<strong><?php esc_html_e( 'Note:', 'woocommerce-square' ); ?></strong>
+								<?php esc_html_e( 'This will only update products that are already synced with Square. New products in Square will not be imported. To import new products, use "Import all Products from Square" on the Settings tab.', 'woocommerce-square' ); ?>
+							</p>
```
As mentioned earlier, this should display only when Square is SOR. This applies to all changes we are making.

iamdharmesh commented on includes/Admin/Sync_Page.php:
```diff
@@ -376,7 +418,7 @@ private static function output_sync_with_square_modal_template() {
 							printf(
 								/* translators: Placeholders: %1$s - the name of the system of record set in the sync settings (e.g. Square or WooCommerce), %3%s - unordered HTML list of additional information item(s) */
-								esc_html__( 'You are about to sync products with Square. %1$s is your system of record set in the sync settings. For all products synced with Square: %2$s', 'woocommerce-square' ),
+								esc_html__( 'You are about to sync existing products with Square. %1$s is your system of record set in the sync settings. For all products currently synced with Square: %2$s', 'woocommerce-square' ),
```
```suggestion
								esc_html__( 'You are about to sync products with Square. %1$s is your system of record set in the sync settings. For all products synced with Square: %2$s', 'woocommerce-square' ),
```

As we are adding notice only, better to keep this as it is or only update if Square is SOR.

Commit:
Refactor sync page messaging for clarity and remove outdated import notice. Updated header and conditional display of sync information to enhance user understanding of product syncing with Square.

iamdharmesh commented on src/new-user-experience/modules/configure-sync/index.js:
```diff
@@ -406,12 +410,35 @@ export const ConfigureSync = ( {
 									<Modal
-										title="Import Products From Square"
+										title={ __(
+											'Import New Products From Square',
```
```suggestion
											'Import Products From Square',
```

iamdharmesh commented on src/new-user-experience/modules/configure-sync/index.js:
```diff
@@ -406,12 +410,35 @@ export const ConfigureSync = ( {
+														'To update existing synced products, use "Sync Now" on the Update tab instead.',
+														'woocommerce-square'
+													) }
+												</p>
```
As [mentioned](https://github.com/woocommerce/woocommerce-square/pull/445#discussion_r2635909887) earlier, Import products option also provide a way to update existing products while doing import, So this is not accurate, please remove this.

iamdharmesh commented on src/new-user-experience/modules/configure-sync/index.js:
```diff
@@ -347,6 +347,10 @@ export const ConfigureSync = ( {
 									className="import-products-wrapper"
+									description={ __(
+										'Use this to bring new products from Square into WooCommerce. This is different from "Sync Now" which only updates products that are already synced.',
+										'woocommerce-square'
```
"This is different from "Sync Now" which only updates products that are already synced." This is correct only when SOR is Square.

iamdharmesh commented on includes/Settings.php:
```diff
@@ -424,6 +424,7 @@ public function generate_import_products_html( $id, $field ) {
 				<a id="wc_square_import_products" href='#' class='button js-import-square-products <?php echo ( ! $this->get_location_id() ? 'disabled' : '' ); ?>'>
 					<?php echo esc_html__( 'Import all products from Square', 'woocommerce-square' ); ?>
 				</a>
+				<p class="description" style="margin-top: 8px;"><?php esc_html_e( 'Use this to bring new products from Square into WooCommerce. This is different from "Sync Now" which only updates products that are already synced.', 'woocommerce-square' ); ?></p>
```
"This is different from "Sync Now" which only updates products that are already synced." This is correct only when SOR is Square.

Commit:
Implement PR review feedback.

Update product import and sync messaging for clarity in WooCommerce Square integration. Enhanced descriptions in the Settings and Sync pages to differentiate between importing new products and syncing existing ones based on the system of record. Improved user guidance for product management.

m-muhsin commented:
Thanks for the kind and detailed feedback, @iamdharmesh. Should look okay now.

iamdharmesh reviewed (approved):
Thanks @m-muhsin.

@m-muhsin Could you please check if PR description is accurate about changes we made in the PR. Also, I think sort description of what we are doing in whole PR would be more helpful instead of AI generate summary with file wise changes. Thank you.

m-muhsin commented:
@iamdharmesh updated PR description accordingly. Thanks!

qasumitbagthariya reviewed (approved):
### QA Update  ✅ 

-------------

I have verified this PR in the `fix/square-225` branch, which has been fixed and is functioning as intended.

I tested the following on this branch:

- The "Sync Now" button now shows helper text: "Updates existing synced products only. 

<img width="1499" height="624" alt="image" src="https://github.com/user-attachments/assets/79ff9902-bf19-443f-87bf-902159b6e7e7" />


- Sync Now" and verify the modal includes a yellow warning note about scope
<img width="1225" height="693" alt="image" src="https://github.com/user-attachments/assets/33d85d1f-7394-445b-811f-1ec2f5487e2c" />


- Import all Products from Square" button has description text below it
<img width="1075" height="599" alt="image" src="https://github.com/user-attachments/assets/4630fd3c-8b62-403e-b46e-af3eba916e3a" />

- Setup Wizard
<img width="1021" height="703" alt="image" src="https://github.com/user-attachments/assets/09264684-8e49-4e39-b42e-d6c7e4f89103" />


**Testing Environment** 

<Details>

- WordPress: 6.9
- Theme: Twenty Twenty-Five 1.4
- WooCommerce - 10.4.0
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS 15.2 
- Branch: fix/square-225

</Details>


Steps to Test- As mentioned in the PR description.
Test Results - It is working as expected.
Functional Demo / Screencast - 
Special Notes -  Ready for UAT
Testing Document status:
Cases related to this Issue/PR are added to the Critical Flow Wiki pages: 

- [ ] Yes
- [x] Not Required/Applicable for this PR
------------------------------------------------

qasumitbagthariya commented:
Regression / Smoke Test Report  ✅ 
-------

Tested with Archive File created via "php woorelease.phar build repo_URL" (Composer version 2.8.4, npm version 10.8.2, node version 20.18.1)

Status- Working expected with Plugin Archive/Zip file same as fix specific branch.

**Testing Environment** 

<Details>

- WordPress: 6.9
- Theme: Storefront 4.6.2
- Theme: Twenty Twenty-Five 1.4
- WooCommerce - 10.4.3
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS 15.2 
- Branch: smoke-testing

</Details>

Next Step- Ready to Merge 🚀
